pkg/virt-controller/watch/volume-migration/volume-migration.go

```diff
@@ -141,6 +147,39 @@ func ValidateVolumes(vmi *virtv1.VirtualMachineInstance, vm *virtv1.VirtualMachi
 			valid = false
 			continue
 		}
+
+		// DataVolumes with a no-csi storage class
+		if v.VolumeSource.DataVolume != nil {
+			dv, err := storagetypes.GetDataVolumeFromCache(vm.Namespace, v.VolumeSource.DataVolume.Name, dvStore)
+			if err != nil || dv == nil {
+				continue
+			}
+
+			// if the dv is in succeeded state then it is safe to use since it has already been populated.
+			if dv.Status.Phase == cdiv1.Succeeded {
+				continue
+			}
+
+			boolUsePopulator := true
+			if usePopulator, ok := dv.Annotations["cdi.kubevirt.io/storage.usePopulator"]; ok {
+				var err error
+				boolUsePopulator, err = strconv.ParseBool(usePopulator)
+				if err != nil {
+					// if the string cannot be parsed correctly, something wrong was set there, then we simply evaluate the
+					// storage class
+					boolUsePopulator = true
+				}
+			}
+			pvc, err := storagetypes.GetPersistentVolumeClaimFromCache(vm.Namespace, dv.Name, pvcStore)
+			if err != nil || pvc == nil {
+				continue
```

Comment:
Similar to the issue above when fetching a DV. 

I think this function should probably return some indication that some DVs/PVCs do not exist and exit the reconcile loop